### PR TITLE
fix(life-cycles): ensure life-cycle preconditions hold

### PR DESCRIPTION
**Pull request by luytena**

### CODE REVIEW

File: addon/components/model-form.hbs [hunk 1]
```diff
@@ -40,6 +40,7 @@
       @onStatusCorrection={{@onStatusCorrection}}
       @getChangeParameters={{@getChangeParameters}}
       @getCorrectionParameters={{@getCorrectionParameters}}
+      @getChangeHint={{@getChangeHint}}
```
Comment (velrest) [inline addon/components/model-form.hbs:43]: This component has definitely grown too big by now :sweat_smile: if we have time in the future we should check if we can split it up somehow.

Comment (luytena) [reply addon/components/model-form.hbs:43]: Yup, it hurt a bit adding yet another field :sweat_smile:

**Comment by velrest** [comment]
Status correction on dwelling is broken from unusable to existing

File: addon/services/building.js [hunk 2]
```diff
@@ -167,11 +150,393 @@ export default class BuildingService extends GwrService {
...
+  async setToApprovedBuilding(
+    transition,
+    cascadeLevel,
+    isDryRun,
+    buildingWork
+  ) {
+    await Promise.all(
+      buildingWork.building.buildingEntrance.map(
+        async (buildingEntrance) =>
+          await Promise.all(
```
Comment (velrest) [inline addon/services/building.js:166]:
```suggestion
        (buildingEntrance) =>
          Promise.all(
```

File: addon/services/construction-project.js [hunk 3]
```diff
@@ -139,4 +752,8 @@ export default class ConstructionProjectService extends GwrService {
       }
     });
   }
+
+  getChangeHint(currentStatus, newStatus) {
+    return ConstructionProject.projectTransitionHint[currentStatus][newStatus];
+  }
```
Comment (velrest) [inline addon/services/construction-project.js:758]: Maybe put this in gwr base service.
```suggestion
  getChangeHint(currentStatus, newStatus) {
    return this.cacheClass.projectTransitionHint[currentStatus][newStatus];
  }
```

File: addon/templates/project/errors.hbs [hunk 4]
```diff
@@ -24,136 +24,138 @@
...
-                @modelDescription={{building.fullAddressTexts
-                }} as |BuildingErrorList|
-              >
-                {{#if buildingWithErrors}}
-                  <UkSwitcher
-                    class="uk-margin-small-top uk-animation-fade" as |switcher|
-                  >
-                    <switcher.nav @type="subnav" as |nav|>
-                      <nav.component uk-switcher @divider={{true}} as |tab|>
+  {{#let (filter-by "building.isNew" false this.project.work) as |work|}}
```
Comment (velrest) [inline addon/templates/project/errors.hbs:27]: Maybe handle this in the controller with something like 
```js
this.project.work.filter(w=>!building.isNew).map(w => w.building).map(b=>({...youobject})
```

File: addon/templates/project/form.hbs [hunk 5]
```diff
@@ -113,31 +116,45 @@
...
             </div>
+          {{/if}}
+          {{#unless this.project.isNew}}
+            {{#if this.project.projectStartDate}}
+              <div class={{spacingClass}}>
+                <Field @inputType="date" @attr="projectStartDate" />
+              </div>
+            {{/if}}
+            {{#if this.project.projectCompletionDate}}
+              <div class={{spacingClass}}>
```
Comment (velrest) [inline addon/templates/project/form.hbs:131]:
```suggestion
              <div class="{{spacingClass}} {{unless this.project.projectCompletionDate "uk-hidden"}}">
```

File: addon/templates/project/linked-buildings.hbs [hunk 6]
```diff
@@ -11,10 +11,26 @@
...
       @modelName="building"
       @removeLink={{this.removeBuildingLink}}
-      @searchRoute="search-building"
-      @searchModels={{array @model}}
-      @newRoute="building.new"
-      @newModels={{array @model}} as |model|
+      @searchRoute={{if
+        (eq this.project.typeOfConstructionProject 6011)
+        "search-building"
+        undefined
```
Comment (velrest) [inline addon/templates/project/linked-buildings.hbs:17]: Is this needed?

**Review by velrest — CHANGES_REQUESTED** [review]

File: addon/controllers/building/edit/dwelling/edit.js [hunk 7]
```diff
@@ -111,6 +140,24 @@ export default class BuildingEditDwellingEditController extends Controller {
     } catch (error) {
       console.error(error);
       this.notification.danger(this.intl.t("ember-gwr.dwelling.saveError"));
+
```
Comment (velrest) [inline addon/controllers/building/edit/dwelling/edit.js:143]: Maybe:
```js
this.intl.t((`ember-gwr.lifeCycles.${error.dwellingId ? "statusErrorDwelling" : "statusErrorBuilding"}`, getError(...context))
```

**Review by velrest — CHANGES_REQUESTED** [review]

**Pull request merged**